tdp_core/security/manager/jwt_security_manager.py

```diff
@@ -100,105 +79,20 @@ def refresh_expiring_jwts(response):
-            # Create a new access token for this user
-            access_token = create_access_token_from_user(user)
-            # Append the access token as cookie
-            set_access_cookies(response, access_token)
-            return response
-
-        @app.route('/logout', methods=['GET', 'POST'])
-        @no_cache
-        def logout():
-            payload, cookies = self.logout()
-            # Create response and add security store payload
-            response = jsonify(msg='Bye Bye', **payload)
-            # Handle cookie changes from the security stores
-            for cookie in cookies:
-                try:
-                    response.set_cookie(**cookie)
-                except Exception:
-                    _log.exception(f'Error setting cookie {cookie} to logout response')
-            # Also unset the access token cookies
-            unset_jwt_cookies(response)
-            return response
-
-        @app.route('/loggedinas', methods=['GET', 'POST'])
-        @no_cache
-        def loggedinas():
-            u = self.current_user
-            if u:
-                _log.debug('user login: ' + u.name)
-                # TODO: Return something interesting
-                return jsonify(name=u.name, roles=u.roles)
-            return '"not_yet_logged_in"'
-
-        # Finally, init every store if any requires that
-        for store in self._user_stores:
-            # Check if the actual "init_app" method is implemented and then call it
-            init_store_app = getattr(store, "init_app", None)
-            if callable(init_store_app):
-                init_store_app(app)
-            # Check if the actual "init_jwt" method is implemented and then call it
-            init_store_jwt = getattr(store, "init_jwt", None)
-            if callable(init_store_jwt):
-                init_store_jwt(self.jwt)
-
-    def login_required(self, fn):
-        @wraps(fn)
-        def decorator(*args, **kwargs):
-            u = self.current_user
-            # Allow access only if a user is available
-            if u:
-                # TODO: Add ability to pass roles=[...], ids=[...], etc. to this decorator. Ideal usage looks like this:
-                # TODO: Think about how stacking of these decorators works...
-                # @login_required(roles=['can_read_views']) or # @login_required(ids=['admin'])
-                return fn(*args, **kwargs)
-            return abort(401)
-        return decorator
+        self._delegate_stores_until_not_none("init_app")
 
     @property
-    def current_user(self) -> PhoveaServerUser:
-        try:
-            return self._load_user_from_request(request)
-        except Exception:
-            _log.exception('Error fetching user from current request')
-            return None
+    def current_user(self) -> Optional[PhoveaServerUser]:
+        with self.app.app_context():  # used for jwt request
+            verify_jwt_in_request()
```

Comment:
If I remember correctly this methd would throw as soon as no token is present. As we want to allow loading users via the other functions as well (like `_api_key_from_headers` or essentially `_delegate_stores_until_not_none`), we should either add [optional=True](https://flask-jwt-extended.readthedocs.io/en/stable/api/#flask_jwt_extended.verify_jwt_in_request) or wrap it in an exception handler, ultimately triggering the other functions.

Reply:
I would prefer the exception handler I think, because an invalid token should not abort if the other methods return a valid user.